draft-thomson-http-oblivious.md

```diff
@@ -210,56 +210,69 @@ Responses are bound to responses and so consist only of AEAD-protected content.
 {{response}} describes the process for constructing and processing an
 Encapsulated Response.
 
+~~~
+Encapsulated Response {
+  Nonce (Nk),
+  AEAD-Protected Response (..),
+}
+~~~
+{: #fig-enc-response title="Encapsulated Response"}
+
+The size of the Nonce field in an Encapsulated Response corresponds to the
+size of an AEAD key for the corresponding HPKE ciphersuite.
+
 ## HPKE Encapsulation of Requests {#request}
 
 Clients encapsulate a request `request` with an HPKE public key `pkR`,
-whose Key Identifier is `keyID` as follows:
+whose wire-encoded Key Identifier is `keyID` as follows:
 
 1. Compute an HPKE context using `pkR`, yielding `context` and encapsulation
    key `enc`.
 
 2. Construct additional associated data, `aad`, by prepending a single byte
-   with a value of 0x01 to the key identifier. The key identifier length is not
+   with a value of 0x01 to the key identifier. The key identifier length is
    included in the AAD.
 
 3. Encrypt (seal) `request` with `keyID` as associated data using `context`,
    yielding ciphertext `ct`.
 
 4. Concatenate the length of `keyID` as a variable-length integer, `keyID`,
-   `enc` and `ct`, yielding an Encapsulated Request `enc_request`.
+   `enc` and `ct`, yielding an Encapsulated Request `enc_request`. Note that 
+   `enc` is of fixed-length, so there is no ambiguity in parsing `enc` and 
+   `ct`.
 
 In pseudocode, this procedure is as follows:
 
 ~~~
 enc, context = SetupBaseS(pkR, "request")
-aad = concat(0x01, keyID)
-ct = context.Seal(aad, request)
-enc_request = concat(vencode(len(keyID)), keyID, enc, ct)
+ct = context.Seal(keyID, request)
+enc_request = concat(keyID, enc, ct)
 ~~~
 
 Servers decrypt an Encapsulated Request by reversing this process. Given an
 Encapsulated Request `enc_request`, a server:
 
 1. Parses `enc_request` into `keyID`, `enc`, and `ct` (indicated using the
    function `parse()` in pseudocode). The server is then able to find the HPKE
-   private key, `skR`, corresponding to `keyID`.
+   private key, `skR`, corresponding to `keyID`. If no such key exists, the 
+   server MUST return an error with HTTP status code 401.
```

Comment:
I'd like to keep the HTTP-layer stuff to its own section.

```suggestion
   private key, `skR`, corresponding to `keyID`. If no key exists, the 
   server returns an error.
```

Reply:
That's fine -- separate issue?